### PR TITLE
Add configuration variables.

### PR DESCRIPTION
**Pull request by WangyangZhou90**
Adding the GC Configurations to logs at the start of the test by calling GC.GetConfigurationVariables().

### CODE REVIEW

File: src/tests/GC/Stress/Framework/ReliabilityFramework.cs [hunk 1]
```diff
@@ -157,9 +157,20 @@ public static int Main(string[] args)
         string configFile = null;
         bool okToContinue = true, doReplay = false;
         string sTests = "tests", sSeed = "seed", exectime = "maximumExecutionTime";
+        string output = "";
```
Comment (mrsharm) [inline src/tests/GC/Stress/Framework/ReliabilityFramework.cs:160]: Making use of StringBuilder here. https://learn.microsoft.com/en-us/dotnet/standard/base-types/stringbuilder

```csharp
StringBuilder sb = new()
...

// To Add..
sb.AppendLine($"{kvp.Key}: {kvp.Value}");

...
// To Log..
rf._logger.WriteToInstrumentationLog(null, LoggingLevels.StartupShutdown, String.Format("Get GC Configuration Variables\n{0}",output.ToString()));
```

Comment (markples) [reply src/tests/GC/Stress/Framework/ReliabilityFramework.cs:160]: or perhaps don't bother with `output` at all and just write the lines one at the time?

Comment (WangyangZhou90) [reply src/tests/GC/Stress/Framework/ReliabilityFramework.cs]: @mrsharm I have updated the source with using StringBuilder.

**Review by mrsharm — APPROVED** [review]
LGTM!

**Pull request merged**